cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/store/ActionCreator.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import { GLOBALS } from 'services/global-constants';
+import { MyPipelineApi } from 'api/pipeline';
+import Store, { Actions } from 'components/PipelineList/DeployedPipelineView/store';
+import { objectQuery } from 'services/helpers';
+import { PROGRAM_STATUSES } from 'services/global-constants';
+import {
+  IPipeline,
+  IPipelineStatus,
+  IStatusMap,
+  IRunsCountMap,
+} from 'components/PipelineList/DeployedPipelineView/types';
+
+enum ProgramType {
+  'cdap-data-pipeline' = 'Workflow',
```

Comment:
nit: Could we use `GLOBALS.etlPipelineTypes` like `[GLOBALS.etlPipelineTypes[0]] :'Workflow'` or better yet we can define the artifacts as ts globals and slowly start using them. Would be useful at some point if the backend changes the name of the artifact.

Reply:
enum cannot use computed property name, so I'm changing this to use an object instead